BankManagerConsoleApp/Model/Bank/BankClerk.swift

```diff
@@ -7,28 +7,28 @@
 
 import Foundation
 
-protocol BankDelegate {
-    func updateWorkData(spendedTime: Double)
+protocol BankDelegate: AnyObject {
+    func updateWorkData()
 }
 
-struct BankClerk {
-    private enum Constant {
-        static let startWorkMessage = "고객 업무 시작"
-        static let finishedWorkMessage = "고객 업무 완료"
-        static let spendingTimeForClient: Double = 0.7
-    }
+private enum Constant {
+    static let startWorkMessageFormat = "%d번 고객 %@업무 시작"
+    static let finishedWorkMessageFormat = "%d번 고객 %@업무 완료"
+}
 
-    private let spendingTimeForClient: Double
+protocol BankClerk {
+    var spendingTimeForClient: Double { get }
+    var delegate: BankDelegate? { get set }
 
-    init(spendingTimeForClient: Double = Constant.spendingTimeForClient) {
-        self.spendingTimeForClient = spendingTimeForClient
-    }
+    func work(client: Client)
+}
 
-    func work(client: Client, delegate: BankDelegate) {
-        print("\(client.waitingNumber) \(Constant.startWorkMessage)")
+extension BankClerk {
+    func work(client: Client) {
+        print(String(format: Constant.startWorkMessageFormat, client.waitingNumber, client.taskType.text))
         let usecondsTimeForAClient = useconds_t(spendingTimeForClient * 1000000)
```

Comment:
TaskType의 프로퍼티로 소요시간을 추가하여 활용하게 된다면 별도의 spendingTimeForClient를 가지지 않아도 될 것 같아요.
```suggestion
        let usecondsTimeForAClient = useconds_t(client.taskType.spendingTime * 1000000)
```